applications/registration-api/src/main/java/no/dcat/controller/DatasetController.java

```diff
@@ -187,27 +193,60 @@ Dataset save(Dataset dataset) {
         logger.info("found old dataset: {}" + oldDataset.getTitle().get("nb"));
 
         JsonObject oldDatasetJson = gson.toJsonTree(oldDataset).getAsJsonObject();
+        List<Concept> conceptListFromReq;
+        List<Concept> conceptsGetByIds = new ArrayList<>();
+        Subject subject = new Subject();
+        List<Subject> subjects = new ArrayList<>();
 
         for(Map.Entry<String, Object> entry : updates.entrySet()) {
             logger.debug("update key: {} value: ", entry.getKey(), entry.getValue() );
             JsonElement changes = gson.toJsonTree(entry.getValue());
+
             if(oldDatasetJson.has(entry.getKey())) {
+
                 oldDatasetJson.remove(entry.getKey());
+
+            }
+            if("concepts".equals(entry.getKey())){
+
+                JsonElement jsonElementConcept = gson.toJsonTree(entry.getValue());
+                Type listType = new TypeToken<List<Concept>>() {}.getType();
+                conceptListFromReq = new Gson().fromJson(jsonElementConcept, listType);
+
+                logger.debug("Concept list size: {}" , conceptListFromReq.size());
+                List<String> ids = conceptListFromReq.stream().map(Concept::getId).collect(Collectors.toList());
+
+                conceptsGetByIds = conceptCatClient.getByIds(ids);
+
+            } else{
+
+                oldDatasetJson.add(entry.getKey(), changes);
             }
-            oldDatasetJson.add(entry.getKey(), changes);
         }
 
         logger.debug("Changed dataset Json element: {}", oldDatasetJson.toString());
 
         Dataset newDataset = gson.fromJson(oldDatasetJson.toString(), Dataset.class);
         newDataset.set_lastModified(Calendar.getInstance().getTime());
 
+        for (Concept concept:  conceptsGetByIds) {
+            subject.setId(concept.getId());
+            subject.setUri(concept.getUri());
+            subject.setDefinition(concept.getDefinition().getText());
+            subject.setPrefLabel(concept.getPrefLabel());
+            subject.setAltLabel(concept.getAltLabel());
+            subject.setIdentifier(concept.getIdentifier());
+            subjects.add(subject);
+        }
+
+        newDataset.setSubject(subjects);
+        newDataset.setConcepts(conceptsGetByIds);
```

Comment:
Hvorfor setter du både concents og subjects? Er det noen grunn til at Concepts er tatt med i tillegg til subjects?

Reply:
ok, skjønner. Standarden for begrep var jo ikke ferdig utviklet da dcat-standarden ble laget.